src/BenchmarkDotNet/Validators/ValidationError.cs

```diff
@@ -50,6 +51,8 @@ public override int GetHashCode()
             }
         }
 
+        internal IEnumerable<ValidationError> AsEnumerable() => new List<ValidationError>() { this };
```

Comment:
this method should not be needed as we can just use `yield` everywhere. Also, if we really needed that it should be returning an array (with the list the caller can cast it to list and add some elements) + array is cheaper than list `new [1] { this }` 

```suggestion
```